utils/scripts/repo-helper-leap15.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+set -uex
+
+# This script is used by dockerfiles to optionally use
+# a local repository instead of a distro provided repository.
+# It will also optionally allow running a /tmp/install script
+# for custom packages if present.
+
+: "${REPO_FILE_URL:=}"
+: "${BASE_DISTRO:=opensuse/leap:15.3}"
+: "${JENKINS_URL:=}"
+: "${REPOS:=}"
+
+disable_repo () {
+    if [ -n "$REPO_FILE_URL" ]; then
+        pushd /etc/dnf/repos.d/
+            mv daos_ci-leap15-artifactory.repo{,.tmp}
+            for file in *.repo; do
+                true > "$file"
+            done
+            mv daos_ci-leap15-artifactory.repo{.tmp,}
+        popd
+    fi
+}
+
+# Use local repo server if present
+# if a local repo server is present and the distro repo server can not
+# be reached, have to bootstrap in an environment to get curl installed
+# to then install the pre-built repo file.
+
+if [ -n "$REPO_FILE_URL" ]; then
+    zypper mr --all --disable
+    zypper addrepo                                                            \
+        "${REPO_FILE_URL%/*/}/opensuse-proxy/distribution/leap/${BASE_DISTRO##*:}/repo/oss/" \
+          temp_opensuse_oss_proxy
+    zypper --non-interactive install curl
+    mkdir -p /etc/zypp/repos.d
+    pushd /etc/zypp/repos.d/
+        curl -k -f -o daos_ci-leap15-artifactory.repo.tmp \
+             "$REPO_FILE_URL"daos_ci-leap15-artifactory.repo
+        for file in *.repo; do
+            true > $file
+        done
+        mv daos_ci-leap15-artifactory.repo{.tmp,}
+    popd
+fi
+zypper --non-interactive --gpg-auto-import-keys install \
+    dnf dnf-plugins-core
+mkdir -p /etc/dnf/repos.d
+pushd /etc/zypp/repos.d/
+    for file in *.repo; do
+        sed -e '/type=NONE/d' < $file > /etc/dnf/repos.d/$file
```

Comment:
(lint) Double quote to prevent globbing and word splitting. [SC2086]
(lint) Double quote to prevent globbing and word splitting. [SC2086]